src/web/handlers/bill.rs

```diff
@@ -171,7 +172,26 @@ pub async fn issue_bill(
         return Err(service::Error::PreconditionFailed);
     }
 
-    let drawer = state.identity_service.get_full_identity().await?;
+    let current_identity = state.get_current_identity().await;
+    let (drawer_public_data, drawer_keys) = match current_identity.company {
+        None => {
+            let identity = state.identity_service.get_full_identity().await?;
+            (
+                IdentityPublicData::new(identity.identity),
+                identity.key_pair,
+            )
+        }
+        Some(company_node_id) => {
```

Comment:
This is just an understanding question: So here we still have a company node_id for now which sort of makes the company addressable on DHT? When we switch to Nostr can we use a message id to have company discovery (for a nostr topic). Until now we added contacts via node_id (and fetched details via dht) how is a company (where I am not part of) going into my contacts?

Reply:
So you're adding a company to the contacts via it's id (which is also its node_id).

But this is separate from the peer_id (libp2p/dht etc.), a company doesn't have a peer_id (it's not on the network on it's own).

So basically, if I'm the signatory of a company, I have to somehow subscribe to updates on that company. With the current (DHT-based) setup, we'll:

* Subscribed to `COMPANY{company_node_id}` on Gossipsub (already implemented)
* Check `BILLS{company_node_id}` for a list of bills (on startup and via request - I'll implement this soon)
  * This is also where an issuer would add a bill for the company

And optimally, we'd find a way to do this coordination via Nostr.

Reply:
OK that means we just have to use a Nostr message I'd instead of a node_id then we can subscribe to that and also deliver metadata if needed. Same for topics other than company. How about privacy? Is this public or encrypted data?

Reply:
In the current implementation it's not encrypted (since it's only temporary), but yes, the messages on that topic would be encrypted with the company pub key (=company_node_id), so it wouldn't need to be private.